compiler+runtime/src/cpp/jank/read/lex.cpp

```diff
@@ -542,27 +733,91 @@ namespace jank::read
               return err(error{ token_start, pos, "unexpected end of real, expecting exponent" });
             }
 
+            if(expecting_more_digits)
+            {
+              ++pos;
+              if(found_r)
+              {
+                return err(error{ token_start,
+                                  pos,
+                                  "unexpected end of radix number, expecting more digits" });
+              }
+              return err(error{
+                token_start,
+                pos,
+                fmt::format("unexpected end of radix {} number, expecting more digits", radix) });
```

Comment:
Why do we have these two different returns? Does the last one not make sense if `found_r` is true?

Reply:
yeah, when found_r is true, we would not know what `radix` is yet until later so the msg could be confusing.